bugbug/tools/agent_tools.py

```diff
@@ -0,0 +1,82 @@
+from dataclasses import dataclass
+from typing import Any
+
+from langchain_core.tools import tool
+from langgraph.runtime import get_runtime
+
+from bugbug.code_search.function_search import FunctionSearch
+from bugbug.code_search.searchfox_api import FunctionSearchSearchfoxAPI
+
+
+@dataclass
+class CodeReviewContext:
+    # TODO: Move the Patch class out of `code_review` to avoid circular imports. Then use it here.
+    patch: Any
+
+
+@tool
+def expand_context(file_path: str, line_number: int) -> str:
+    """Expand the context around a specific line in a file diff.
+
+    Args:
+        file_path: The path to the file.
+        line_number: The line number to expand context around. It should be based on the original file, not the patch.
+
+    Returns:
+        The expanded context.
+    """
+    runtime = get_runtime(CodeReviewContext)
+    file_content = runtime.context.patch.get_old_file(file_path)
+
+    def get_file(commit_hash, path):
+        assert path == file_path
+        assert commit_hash == "N/A"
+        return file_content
+
+    # TODO: Implement the logic to for expanding the context using an AST parser
+    # like tree-sitter instead of relying on Searchfox.
+    functions = FunctionSearchSearchfoxAPI(get_file).get_function_by_line(
+        # FIXME: We do not use the commit hash here. Using a placeholder for now.
+        "N/A",
+        file_path,
+        line_number,
+    )
+
+    if not functions:
+        # Fallback: return a few lines around the line number.
+        lines = file_content.splitlines()
+        start = max(0, line_number - 20)
+        end = min(len(lines), line_number + 20)
+        return "\n".join(lines[start:end])
+
+    return functions[0].source
```

Comment:
Can't we just use the file contents from Phabricator here always, without hitting Searchfox?

Reply:
Done here: https://github.com/mozilla/bugbug/compare/e654b3b979e813fb6745374cf09bea596a2146b5..2d31294119c92ccbf9faf7ab69e54bd69ad21910